src/server/Commands.hs

```diff
@@ -206,6 +210,114 @@ cmdPasv sock state params
       sendLine sock "530 Log in first."
       return state
 
+-- cmdQuit handles the QUIT command
+-- takes: controlSocket, UserState
+-- Closes the control connection
+cmdQuit :: Socket -> UserState -> IO UserState
+cmdQuit sock state =
+  do
+    sendLine sock "221 Service closing control connection."
+    close sock
+    return state
+
+-- cmdNlst handles the NLST command
+-- takes: controlSocket, UserState, and a parameter list provided by the client.
+-- Sends directory listing from server to client.
+cmdNlst :: Socket -> UserState -> [String] -> IO UserState
+cmdNlst sock state params
+  | getIsLoggedin state =
+    case () of
+      ()
+        | getDataSock state == NoConnection ->
+          do
+            sendLine sock "425 Data connection was not established."
+            return state
+        | 'l' `notElem` getPerms state -> -- Check if the user is allowed to invoke NLST
+          do
+            sendLine sock "550 Requested action not taken. Not Allowed."
+            return state
+        | otherwise -> do
+          case () of
+            ()
+              | checkParams params 0 ->
+                do
+                  let dataSock = getSock (getDataSock state)
+                  listFiles sock dataSock (getRootDir state)
+                  return state
+              | checkParams params 1 ->
+                do
+                  let dataSock = getSock (getDataSock state)
+                  let pathName = getFirst params
+                  listFiles sock dataSock pathName
+                  return state
```

Comment:
```suggestion
                  if isSubdirectory rootDir pathName
                  then do 
                    listFiles sock dataSock pathName
                    return state
                  else do
                    sendLine sock "550 Requested action not taken. Not Allowed."
                    return state
```
Check if the specified path is within the rootDir
use [isSubdirectory](https://github.com/selbetar/FTPServer/pull/11/files#diff-70bda53810ba0df9e5e078dff520fc05e995962ca2474fa1d97d61c9e0f8f4c7R411) function from PR #11